draft-ietf-quic-transport.md

```diff
@@ -3274,6 +3274,22 @@ The considerations for processing ICMP messages in the previous section also
 apply if these messages are used by DPLPMTUD.
 
 
+### PMTU Probes Containing Source Connection ID
+
+Endpoints that rely on the destination connection ID for routing QUIC packets
+are likely to require that connection ID included in PMTU probe packets in order
+to route resulting ICMP messages ({{icmp-pmtud}}) back to the correct endpoints.
+
+Only long header packets ({{long-header}}) contain source connection IDs, but
+long header packets will not be acknowledged once the connection has been
+established.  One way to construct a PMTU probe is to coalesce (see
+{{packet-coalesce}}) a Handshake packet ({{packet-handshake}}) with a short
```

Comment:
It might make sense to:
* Call the datagram "a probe", rather than referring to the short header packet as "a probe".
* it might be better to suggest use of _any_ coalescible long header packet rather than specifically suggesting the use of a Handshake packet.

Therefore something like: _One way to construct a PMTU probe is to coalesce a Long header packet with a Length field and a short header packet into a single UDP datagram, If the datagram reaches the endpoint, the short header packet will be acknowledged. If the datagram elicits an ICMP message, that message will likely contain the source connection ID within the quoted portion of the datagram._

Reply:
> it might be better to suggest use of any coalescible long header packet rather than specifically suggesting the use of a Handshake packet.

Here are the long header packet types:

1. Initial.
I hesitate to use _initial_, because it is plaintext (no decryption failures) and has an ability to actually create new connections due to unfortunate delays / mis-routing.  Load balancer infrastructure may treat them differently from other packets.

2. 0-rtt.
These are supposed to be client-to-server only.  Having servers send them may be a bit unexpected.

3.  Handshake.
That's what I picked.

4.  Retry.
This does not have length, so it is not eligible.

That leaves only Handshake as the best candidate.  And one type is enough.

Reply:
@igorlord I am fine with using a Handshake packet considering that this is an example (i.e., "One  way of...").

Though personally, I might be interested in sending an 0-RTT packet with zero-length payload (i.e. the Length field set to zero) even just to grease the client implementations.